collector/powersupplyclass.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nopowersupplyclass
+// +build linux
+
+package collector
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs/sysfs"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	powerSupplyClassIgnoredPowerSupplies = kingpin.Flag("collector.powersupply.ignored-supplies", "Regexp of power supplies to ignore for powersupplyclass collector.").Default("^$").String()
+)
+
+type powerSupplyClassCollector struct {
+	subsystem      string
+	ignoredPattern *regexp.Regexp
+	metricDescs    map[string]*prometheus.Desc
+}
+
+func init() {
+	registerCollector("powersupplyclass", defaultEnabled, NewPowerSupplyClassCollector)
+}
+
+func NewPowerSupplyClassCollector() (Collector, error) {
+	pattern := regexp.MustCompile(*powerSupplyClassIgnoredPowerSupplies)
+	return &powerSupplyClassCollector{
+		subsystem:      "power_supply",
+		ignoredPattern: pattern,
+		metricDescs:    map[string]*prometheus.Desc{},
+	}, nil
+}
+
+func (c *powerSupplyClassCollector) Update(ch chan<- prometheus.Metric) error {
+	powerSupplyClass, err := getPowerSupplyClassInfo(c.ignoredPattern)
+	if err != nil {
+		return fmt.Errorf("could not get power_supply class info: %s", err)
+	}
+	for _, powerSupply := range powerSupplyClass {
+
+		for name, value := range map[string]*int64{
+			"authentic":                   powerSupply.Authentic,
+			"calibrate":                   powerSupply.Calibrate,
+			"capacity":                    powerSupply.Capacity,
```

Comment:
Do we know what unit these metrics are? Watt-hours? Amp-Hours?

Reply:
`capacity` is a percentage value.